File: julia/robot-name/robot-name.jl
using Random
rl = Set()
rc = Set()

mutable struct Robot
	name::String
	Robot() = reset!(new())
end

function reset!(instance::Robot)
	instance.name = name(instance)
	delete!(rl, instance.name)
	push!(rc, instance.name)
	return instance
end

function name(r::Robot)
	name = join(randstring('A':'Z', 2)*randstring('0':'9', 3))
	while (name in rl || name in rc)
		name = join(randstring('A':'Z', 2)*randstring('0':'9', 3))
	end
	r.name = name
	push!(rl, r.name)
	return r.name
end

File: python/robot-name/robot_name.py
import random
import string

RobotLibrary = set()

class Robot:

    def __init__(self):
        self.name = self.generate_name()
   
    def generate_name(self):
        letters = random.choices(string.ascii_uppercase, k=2)
        numbers = random.choices(string.digits, k=3)
        name = ''.join(letters + numbers)
        if name not in RobotLibrary:    
           RobotLibrary.add(name)
           return name
   
    def reset(self):
        self.name = self.generate_name()
